src/Select.js

```diff
@@ -307,6 +328,16 @@ var Select = React.createClass({
 		if (this.props.disabled || (event.type === 'mousedown' && event.button !== 0)) {
 			return;
 		}
+		
+		if (this.props.maxMultiSelection > 0 && this.state.isOpen) {
+			if (this.state.values.length + 1 > this.props.maxMultiSelection && this.props.replaceIfMax) {
```

Comment:
I think it would be better to combine both cases onto a single `if (a or b)` to avoid duplicating the code and potential inconsistencies...the condition would be pretty long as it is, so if gets hard to read maybe storing them on 2 different variables?